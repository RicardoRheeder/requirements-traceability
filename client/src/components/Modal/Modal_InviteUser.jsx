import React from 'react'
import { useSelector } from 'react-redux'

export default function Modal_InviteUser() {
  const modalObject = useSelector((state) => state.common.modalObject, [])

  return (
    <div className="modal-root modal-root-child">
      <div className="modal-contents-container">
        <h2>{'Invite user(s) to document'}</h2>
        <div>
          {'Please enter the email address(es) of the user(s) to invite.'}
        </div>
        <div>
          {'When entering multiple email addresses, use commas to separate.'}
        </div>
        <form>
          <input
            className="modal-input"
            type="email"
            id="emails"
            name="emails"
            multiple
          />
          <button className="orange-button submit-button">Submit</button>
        </form>
      </div>
    </div>
  )
}
