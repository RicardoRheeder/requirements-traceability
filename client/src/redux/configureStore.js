import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

// restricting extension in production using process.env.NODE_ENV which is set 'production'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction';

import store from './stores';

const enhancer = (process.env.NODE_ENV == 'development')
               ? composeWithDevTools(applyMiddleware(thunk))
               : applyMiddleware(thunk);

export default () => createStore(store, enhancer);
