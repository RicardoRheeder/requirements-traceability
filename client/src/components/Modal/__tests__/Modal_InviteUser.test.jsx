import React from "react"
import Enzyme, { shallow, mount} from "enzyme"
import Modal_InviteUser from "../Modal_InviteUser"
import Adapter from "enzyme-adapter-react-16"
import { Provider } from "react-redux";
import configureStore from "redux-mock-store"


const mockStore = configureStore([])
Enzyme.configure({adapter: new Adapter() });

describe("invite user modal component", () => {
    let store
    beforeEach(() => {
        store = mockStore({common:{
            modalObject: { visible: true},
            selectedDocumentPanelObject: {_id:"test"}
        },
        user: {info: "testinfo"}});
    });
    test("renders", () => {
        const wrapper = shallow(<Provider store = {store}><Modal_InviteUser/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
    test("renders with children", () => {
        const wrapper = mount(<Provider store = {store}><Modal_InviteUser/></Provider>);
        expect(wrapper.exists()).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });
});