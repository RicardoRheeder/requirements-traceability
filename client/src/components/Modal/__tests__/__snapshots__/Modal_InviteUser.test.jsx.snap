// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invite user modal component renders 1`] = `ShallowWrapper {}`;

exports[`invite user modal component renders with children 1`] = `ReactWrapper {}`;
